docs/design/README.md

```diff
@@ -0,0 +1,552 @@
+For this walkthrough, we will use the directory structure shown above.
+This means we will have the following items and dependencies:
+* Variant: `example-dev` which depends on the `hello-dev-kit` by placing it in the `Cargo.toml` `build-dependencies` section.
+* Kit: The `hello-dev-kit` depends on the `hello-agent` package by placing it in the `Cargo.toml` `build-dependencies` section.
+* Package: The `hello-agent` package.
+
+The user kicks things off with:
+
+```sh
+twoliter build --variant example-dev --arch aarch64
+```
+
+First Twoliter learns the external kit dependency graph using the traversal described for `twoliter update`.
+It pulls all kit container images that will be needed for the build.
+
+Twoliter starts the twoliter container environment and mounts the project directory.
+The container has a copy of Bottlerocket's `Makefile.toml` build tools in it.
+Twoliter invokes the `cargo make build` task.
+
+The `cargo make build` task is an alias of `cargo make variant` and its entrypoint is the variant's `Cargo.toml` file.
+The variant's `Cargo.toml` will specify needed local kits in its `build-dependencies` section.
+This means that Cargo will build these kits before proceeding with the variant build.
+
+Before the variant `build.rs` runs, the kit needs to be built.
+Before the kit `build.rs` runs, the package needs to be built.
+
+So the first thing `cargo` will do is run the `hello-agent` package's `build.rs`.
+This will launch `buildsys build-package`.
+A container build will create the `hello-agent` RPM file.
+
+After that, the kit's `build.rs` will run.
+This will launch `buildsys build-kit`.
+A container build will run creating the container image `hello-dev-kit`.
+
+Once the local kit has been built, the variant's `build.rs` can run.
+This will launch `buildsys build-variant`.
+
+Before the variant can be built, a composite of all the required local and external kits needs to be created.
+Buildsys will create a container with a directory structure like this:
+
+```
+❯ tree / --dirsfirst
+.
+├── etc
+│   └── yum.repos.d
+│       ├── bottlerocket-core-kit.repo
+│       ├── ...other bottlerocket core project kits
+│       └── hello-dev-kit.repo
+└── kits
+    ├── bottlerocket-core-kit
+    │   └── ...rpms
+    ├── ...other bottlerocket core project kits
+    │   └── ...rpms
+    └── hello-dev-kit
+        └── ...rpms
+```
+
+The kit priority order will be provided in the variant's `Cargo.toml` metadata section.
+Under the hood, Twoliter will composite the kits with dnf configurations for repo priority.
+
+After compositing the kits, the variant image build will proceed similarly to the way that it used to.
+Before installing the RPM packages, the Bottlerocket SDK's Fedora-provided `/etc/yum.repos.d/` directory will be occluded by the kits composite.
+This way, dnf install commands will pull from the kit-provided RPM repositories.
+If a package is available in more than one kit, it will be taken from the higher priority kit by dnf.
```

Comment:
Will the user get any notification when a package is present in two different kits? What if one of kits used by the users introduced a package present in an existing kit and the user doesn't realize that until they run their image?